src/riak_kv_index_fsm.erl

```diff
@@ -78,10 +78,15 @@ use_ack_backpressure() ->
 
 %% @doc Construct the correct index command record.
 -spec req(binary()|tuple(binary()), term(), term()) -> term().
-req(Bucket, _ItemFilter, ?SQL_SELECT{} = Q) ->
-    #riak_kv_sql_select_req_v1{bucket=Bucket,
-                               qry=Q};
 req(Bucket, ItemFilter, Query) ->
```

Comment:
This would then have to be modified for every version increment, it was a real surprise to me that this record made it into riak_kv_index_fsm. If we keep the checks in riak_kv_select the changes will be isolated.